homeassistant/components/synology_dsm/media_source.py

```diff
@@ -0,0 +1,240 @@
+            try:
+                albums = await diskstation.api.photos.get_albums()
+            except SynologyDSMException:
+                return []
+
+            ret = []
+            ret.append(
+                BrowseMediaSource(
+                    domain=DOMAIN,
+                    identifier=f"{item.identifier}/0",
+                    media_class=MediaClass.DIRECTORY,
+                    media_content_type=MediaClass.IMAGE,
+                    title="All images",
+                    can_play=False,
+                    can_expand=True,
+                )
+            )
+            for album in albums:
+                ret.append(
+                    BrowseMediaSource(
+                        domain=DOMAIN,
+                        identifier=f"{item.identifier}/{album.album_id}",
+                        media_class=MediaClass.DIRECTORY,
+                        media_content_type=MediaClass.IMAGE,
+                        title=album.name,
+                        can_play=False,
+                        can_expand=True,
+                    )
+                )
+
+            return ret
+
+        # Request items of album
+        # Get Items
+        # The library works sync, and this expects async calls
+        album = SynoPhotosAlbum
+        album.album_id = int(identifier_parts[1])
+        try:
+            items = await diskstation.api.photos.get_items_from_album(album, 0, 1000)
+        except SynologyDSMException:
+            return []
+
+        ret = []
+        for items_item in items:
+            mime_type, _ = mimetypes.guess_type(items_item.file_name)
+            assert isinstance(mime_type, str)
+            parts = mime_type.split("/")
+            if parts[0] == "image":
+                ret.append(
+                    BrowseMediaSource(
+                        domain=DOMAIN,
+                        identifier=f"{identifier_parts[0]}/{identifier_parts[1]}/{items_item.thumbnail_cache_key}/{items_item.file_name}",
+                        media_class=MediaClass.IMAGE,
+                        media_content_type=mime_type,
+                        title=items_item.file_name,
+                        can_play=True,
+                        can_expand=False,
+                        thumbnail=await self.async_get_thumbnail(
+                            items_item.thumbnail_cache_key,
+                            items_item.item_id,
+                            "sm",
+                            identifier_parts[0],
+                        ),
+                    )
+                )
+        return ret
+
+    async def async_resolve_media(self, item: MediaSourceItem) -> PlayMedia:
+        """Resolve media to a url."""
+        parts = item.identifier.split("/")
+        cache_key = parts[2]
+        mime_type, _ = mimetypes.guess_type(parts[3])
+        assert isinstance(mime_type, str)
+        return PlayMedia(f"/synology_dsm/{parts[0]}/{cache_key}/{parts[3]}", mime_type)
```

Comment:
Do we know that parts will always have all of these indices? The code here doesn't make it clear what exactly we're splitting apart.

Reply:
Yes those parts will always be there, because that is how I build the folder structure. But I added them as variables now.
Take a look if you agree